buzzmobile/process/gps_mapper/interpolate.py

```diff
@@ -0,0 +1,151 @@
+import numpy as np
+import math
+import cv2
+
+def interpolate(points, sigma_x, sigma_y, height=500, width=500):
+    """
+    Takes a list of points, connects them with a line,
+    and Gaussian blurs the line.
+    ----------------------------------------------------------------------------
+    points: list of tuples, i.e. (x, y) points that will be plotted on the image
+    sigma_x, sigma_y: gaussian kernel paramters (higher = more blurry)
+    height, width: dimensions of the output image in pixels
+    ----------------------------------------------------------------------------
+    """
+    output = np.zeros((height, width), np.uint8)
+    x_vals = [x for (x, y) in points]
+    y_vals = [y for (x, y) in points]
+    left = min(x_vals)
+    bottom = max(y_vals)
+    for i in range(len(points) - 1):
+        pt1 = points[i]
+        pt2 = points[i+1]
+        cv2.line(output, pt1, pt2, [255, 255, 255], 4)
+    output = cv2.GaussianBlur(output, (sigma_x, sigma_y), 0)
+    return output
+
+def haversine(lat1, lon1, lat2, lon2):
+    """
+    Calculates the distance between two lat-lon points in kilometers.
+    """
+    delta_lat = math.radians(lat2 - lat1)
+    delta_lon = math.radians(lon2 - lon1)
+    lat1 = math.radians(lat1)
+    lat2 = math.radians(lat2)
+    a = math.sin(delta_lat / 2)**2 + math.cos(lat1) * math.cos(lat2) * math.sin(delta_lon/2)**2
+    c = 2 * math.atan2(math.sqrt(a), math.sqrt(1 - a))
+    R = 6371
+    d = R * c
+    return d
+
+def normalized_points(points, height=500, width=500):
+    """
+    Takes a list of points (tuples of x, y coordinates) and an output image size
+    and returns a transformed list of points that fit in the output image.
+    """
+    top_left = (min([x for (x, y) in points]), min([y for (x, y) in points]))
+    bottom_right = (max([x for (x, y) in points]), max([y for (x, y) in points]))
+    x_range = abs(top_left[0] - bottom_right[0])
+    y_range = abs(top_left[1] - bottom_right[1])
+    return [((x - top_left[0]) * width / x_range, 
+        (y - top_left[1]) * height / y_range) 
+        for (x, y) in points]
+    
+def normalize_single_point(y_range, x_range, height, width, top_left, bottom_right, point):
+    return ((point[0] - top_left[0]) * width / x_range, (point[1] - top_left[1]) * height / y_range)
+
+
+def window(image, location, angle, height=500, width=500):
```

Comment:
these are defaults, they get called from gps_mapper.py so it will get changed to use parameters file in gps_mapper.py